### PR TITLE
use more robust project.exists function

### PR DESCRIPTION
**Pull request by girishramnani**
## What is the purpose of this change? What does it change?

Currently to check if a project exists or not we used an inefficient List All projects -> match the name approach. This approach becomes quite slow when large number of projects are present. 
Also the `project.Exists` is used at many places and many times which multiplies the slowness. 
This PR addresses this issue by providing a more robust approach of just getting the project.

In my primary cluster which has very less projects I got a 5-7% increase in speed in `project` related commands.

## Was the change discussed in an issue?
This PR possibly resolves the below issue
fixes https://github.com/openshift/odo/issues/1717

## How to test changes?
`odo project create <name> #twice should fail`

### CODE REVIEW

**Comment by kadel** [comment]
/retest
```
[odo]  ✗  Unable to connect to OpenShift cluster, is it down?
```
login error

**Comment by surajnarwade** [comment]
travis is failing

**Comment by girishramnani** [comment]
login failure 
/retest

**Comment by girishramnani** [comment]
again login error
/retest

File: pkg/occlient/occlient.go [hunk 1]
```diff
@@ -470,6 +471,25 @@ func (c *Client) GetProjectNames() ([]string, error) {
 	return projectNames, nil
 }
 
+func (c *Client) GetProject(projectName string) (*projectv1.Project, error) {
+	prj, err := c.projectClient.Projects().Get(projectName, metav1.GetOptions{})
+	if err != nil {
+		istatus, ok := err.(kerrors.APIStatus)
+		if ok {
+			status := istatus.Status()
+			// the
```
Comment (kadel) [inline pkg/occlient/occlient.go]: this comment can probably be deleted ;-)

Comment (girishramnani) [reply pkg/occlient/occlient.go]: yeah, will do

Comment (girishramnani) [reply pkg/occlient/occlient.go]: done

**Comment by girishramnani** [comment]
e2e-core succeeded twice in a row @kadel

File: pkg/occlient/occlient.go [hunk 2]
```diff
@@ -470,6 +471,24 @@ func (c *Client) GetProjectNames() ([]string, error) {
 	return projectNames, nil
 }
 
+func (c *Client) GetProject(projectName string) (*projectv1.Project, error) {
```
[the remark this answers is not in the document]
Comment (mohammedzee1000) [reply pkg/occlient/occlient.go]: @girishramnani ^

**Comment by girishramnani** [comment]
login failure
/retest

**Pull request merged**